js/king.js

```diff
@@ -0,0 +1,53 @@
+import {Piece} from "./piece.js";
+
+export class King extends Piece {
+    /**
+     * Constructor: inherits from abstract Piece class
+     * @param id piece id
+     * @param rowPos current row position
+     * @param colPos current column position
+     * @param playerId player id
+     */
+    constructor(id, rowPos, colPos, playerId) {
+        super();
+        this.id = id;
+        this.rowPos = rowPos;
+        this.colPos = colPos;
+        this.playerId = playerId;
+        this.unicodeChar = "&#9812;";
+    }
+
+    /**
+     * Gets a list of possible moves the rook piece can make including attack moves
+     * @param board Board singleton instance
+     * @returns {*[]}
+     */
+    possibleMoves(board) {
+        let possibleMoves = [];
+        if (this.rowPos + 1 <= 7 && board[this.rowPos + 1][this.colPos].getPlayerId() !== this.playerId) {  // check below
```

Comment:
The condition for every `if` statement will throw an error when attempting to use the index operator `[]` on `null` values in the board.